src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutes.scala

```diff
@@ -124,14 +124,23 @@ trait ResourceRoutes extends UserInfoDirectives with SecurityDirectives with Sam
 
   def postResource(resourceType: ResourceType, userInfo: UserInfo): server.Route =
     post {
-      entity(as[CreateResourceRequest]) { createResourceRequest =>
-        if (resourceType.reuseIds && resourceType.isAuthDomainConstrainable) {
-          throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "this api may not be used for resource types that allow both authorization domains and id reuse"))
-        }
-        complete(
-          resourceService
-            .createResource(resourceType, createResourceRequest.resourceId, createResourceRequest.policies, createResourceRequest.authDomain, userInfo.userId)
-            .map(_ => StatusCodes.NoContent))
+        entity(as[CreateResourceRequest]) { createResourceRequest =>
+          if (resourceType.reuseIds && resourceType.isAuthDomainConstrainable) {
+            throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "this api may not be used for resource types that allow both authorization domains and id reuse"))
+          }
+          complete(
+            createResourceRequest.returnResource match {
+              case true =>
+                StatusCodes.Created -> resourceService
+                  .createResource(resourceType, createResourceRequest.resourceId, createResourceRequest.policies, createResourceRequest.authDomain, userInfo.userId)
+                  .map { r => CreateResourceResponse(r.resourceTypeName, r.resourceId, r.authDomain, r.accessPolicies.map(ap => CreateResourcePolicyResponse(ap.id, ap.email))) }
+              case _ =>
+                resourceService
+                  .createResource (resourceType, createResourceRequest.resourceId, createResourceRequest.policies, createResourceRequest.authDomain, userInfo.userId)
+                  .map { _ => StatusCodes.NoContent }
+            }
+          )
+
```

Comment:
instead of copy/pasting the `createResource(...)` call in both cases, could you do that once? Something like:
```
resourceService.createResource(...).map{ r =>
  if (createResourceRequest.returnResource)
    StatusCodes.Created -> CreateResourceResponse(r. ...)
  else
    StatusCodes.NoContent
}
```

note also for personal style I'm using `if` instead of `match`ing on a boolean; your call on style here.

( I did not test this suggestion; I'm coding right in github )

Reply:
tried -- couldn't get it to go (implicit marshaller related?) . If you have some time to help make it work, I'm game

Reply:
this compiles for me, haven't run tests on it:
```
          val resourceMaker: Future[ToResponseMarshallable] = resourceService
            .createResource(resourceType, createResourceRequest.resourceId, createResourceRequest.policies, createResourceRequest.authDomain, userInfo.userId)
              .map { r =>
                if (createResourceRequest.returnResource.contains(true)) {
                  StatusCodes.Created -> CreateResourceResponse(r.resourceTypeName, r.resourceId, r.authDomain, r.accessPolicies.map(ap => CreateResourcePolicyResponse(ap.id, ap.email)))
                }  else {
                  StatusCodes.NoContent
                }
          }

          complete(resourceMaker)
```

Reply:
note I'm using  `createResourceRequest.returnResource.contains(true)` ... I'm not convinced `returnResource` needs to be an `Option`, so you might do away with the `contains` too.

Reply:
updated -- I think it needs to be an option for the akka-http binding/parsing to work properly, Matt saw some errors along these lines earlier